src/main/java/org/jabref/gui/preferences/PreferencesSearchHandler.java

```diff
@@ -0,0 +1,140 @@
+package org.jabref.gui.preferences;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import javafx.beans.property.ListProperty;
+import javafx.beans.property.SimpleListProperty;
+import javafx.beans.property.StringProperty;
+import javafx.collections.FXCollections;
+import javafx.css.PseudoClass;
+import javafx.scene.Node;
+import javafx.scene.Parent;
+import javafx.scene.control.Labeled;
+
+import com.google.common.collect.ArrayListMultimap;
+
+class PreferencesSearchHandler {
+
+    private static PseudoClass labelHighlight = PseudoClass.getPseudoClass("search-highlight");
+    private final List<PrefsTab> preferenceTabs;
+    private final StringProperty searchText;
+    private final ListProperty<PrefsTab> filteredPreferenceTabs;
+    private final ArrayListMultimap<PrefsTab, LabeledWrapper> preferenceTabsLabelNames;
+    private final ArrayList<LabeledWrapper> highlightedLabels = new ArrayList<>();
+
+    private class LabeledWrapper {
+
+        private final Labeled labeled;
+
+        LabeledWrapper(Labeled _labeled) {
```

Comment:
Please do not use an underscore. In java it's more common to do:
```suggestion
        LabeledWrapper(Labeled labeled){
        this.labeled= labeled;
```

Reply:
And maybe use a better name to indicate what the purpose of the class actually is. LabeledWrapper sounds a bit too generic for me.

Reply:
Initial idea with the inner class was to handle `Labeled` being treated as a key to mapping from `Labeled` to `PrefsTab`. Now that the relation is reversed, I don't think the `LabeledWrapper` class is required anymore.